perses/rjmc/geometry.py

```diff
@@ -666,670 +705,21 @@ def _torsion_logp(self, atom, xyz, torsion, atoms_with_positions, positions, bet
             bond_atom = torsion.atom3
             angle_atom = torsion.atom2
             torsion_atom = torsion.atom1
-        involved_angles = self._get_valid_angles(atoms_with_positions, atom)
-        involved_torsions = self._get_torsions(atoms_with_positions, atom)
         internal_coordinates = self._autograd_ctoi(xyz, positions[bond_atom.idx], positions[angle_atom.idx], positions[torsion_atom.idx])
-        if not Z:
-            p, Z = self._normalize_torsion_proposal(atom, internal_coordinates[0], internal_coordinates[1], bond_atom, angle_atom, torsion_atom, atoms_with_positions, positions, beta, n_divisions=5000)
-        ub_torsion = self._torsion_and_angle_potential(xyz, atom, positions, involved_angles, involved_torsions, beta)
-        p_torsion = np.exp(-ub_torsion) / Z
-        return p_torsion
+        p, Z, q, phis, logq = self._normalize_torsion_proposal(atom, internal_coordinates[0], internal_coordinates[1], bond_atom, angle_atom, torsion_atom, atoms_with_positions, positions, beta, n_divisions=5000)
+        #find the phi that's closest to the internal_coordinate phi:
+        phi_idx, phi = min(enumerate(phis), key=lambda x: abs(x[1]-internal_coordinates[2]))
+        p_phi = p[phi_idx]
+        return np.log(p_phi)
```

Comment:
`log(p_phi)` is not numerically robust.  We want to return 
```python
logq[phi_idx] - logSumExp(logq)
```
instead, which is numerically reobust.

In fact, why does `_normalize_torsion_proposal` return `logq` instead of `logp`?